photon-api/src/main/scala/com/linkedin/photon/ml/data/LocalDataset.scala

```diff
@@ -188,61 +188,69 @@ object LocalDataset {
   protected[ml] def computePearsonCorrelationScore(
       labelAndFeatures: Array[(Double, Vector[Double])]): Map[Int, Double] = {
 
-    val featureLabelProductSums = mutable.Map[Int, Double]()
-    val featureFirstOrderSums = mutable.Map[Int, Double]()
-    val featureSecondOrderSums = mutable.Map[Int, Double]()
-    var labelFirstOrderSum = 0.0
-    var labelSecondOrderSum = 0.0
-    var numSamples = 0
+    val featureMeans = mutable.Map[Int, Double]()
+    var featureUnscaledVars = mutable.Map[Int, Double]()
+    var labelMean = 0.0
+    var labelUnscaledVar = 0.0
+    val unscaledCovariances = mutable.Map[Int, Double]()
     var interceptAdded = false
+    var numSamples = 0
 
     labelAndFeatures.foreach { case (label, features) =>
       numSamples += 1
-      labelFirstOrderSum += label
-      labelSecondOrderSum += label * label
+
+      val deltaLabel = label - labelMean
+      labelMean += deltaLabel / numSamples
+      labelUnscaledVar += deltaLabel * (label - labelMean)
+
       // Note that, if there is duplicated keys in the feature vector, then the following Pearson correlation scores
       // calculation will screw up
       features.activeIterator.foreach { case (key, value) =>
-        featureFirstOrderSums.update(key, featureFirstOrderSums.getOrElse(key, 0.0) + value)
-        featureSecondOrderSums.update(key, featureSecondOrderSums.getOrElse(key, 0.0) + value * value)
-        featureLabelProductSums.update(key, featureLabelProductSums.getOrElse(key, 0.0) + value * label)
+        val prevFeatureMean = featureMeans.getOrElse(key, 0.0)
+        val deltaFeature = value - prevFeatureMean
+        val featureMean = prevFeatureMean + deltaFeature / numSamples
+
+        val prevFeatureUnscaledVar = featureUnscaledVars.getOrElse(key, 0.0)
+        val featureUnscaledVar = prevFeatureUnscaledVar + (value - prevFeatureMean) * (value - featureMean)
```

Comment:
You can use `deltaFeature` to replace `(value - prevFeatureMean)`.

Reply:
Will do!